### DISCUSSION TITLE
Your report could not be applied to any marketplaces!!!

### DISCUSSION THREAD

**Discussion by smilekudo** [saleweaver/python-amazon-sp-api#51] [Q&A]
Hi, i got a script to run and it showed me this issue.
Where should i fixed?
I tried with UK and GB also same problem.

```
import datetime
from sp_api import api
from sp_api import base
from sp_api import auth
credentials=dict(
        refresh_token='',
        lwa_app_id='xxx',
        lwa_client_secret='xxx',
        aws_secret_key='xxx',
        aws_access_key='xxx',
        role_arn='xxx'
    )

# report request     
createReportResponse = api.Reports(credentials=credentials, marketplace=base.Marketplaces.UK).create_report(reportType='GET_FLAT_FILE_OPEN_LISTINGS_DATA')
```

This is issue:
`sp_api.base.exceptions.SellingApiBadRequestException: [{'code': 'InvalidInput', 'message': 'Report rejected for this client. your report could not be applied to any marketplaces', 'details': ''}]`

**Comment by saleweaver** [comment]
I've never seen this error, sorry. No idea. Are you suspended in the marketplace?

**Reply by smilekudo** [reply]
Is that code format is correct? @saleweaver

**Reply by fkhjoy** [reply]
I also got the same issue while creating report.

**Reply by saleweaver** [reply]
@smilekudo Do you get this error in tests or production?

**Reply by smilekudo** [reply]
Production, i ran it on java but python not work

**Reply by saleweaver** [reply]
I've found this on Amazon's repo. I'm getting the error too at the moment

**Reply by saleweaver** [reply]
@smilekudo Can you check if this one is working for you:

```
    createReportResponse = Reports(marketplace=Marketplaces.US).create_report(
        reportType='GET_XML_RETURNS_DATA_BY_RETURN_DATE', marketplaceIds=[Marketplaces.US.marketplace_id])
```

**Reply by smilekudo** [reply]
@saleweaver : yes worked.
Here example for who are looking for:
```
import datetime
from sp_api import api
from sp_api import base
from sp_api import auth
credentials=dict(
        refresh_token='xxxxx',
        lwa_app_id='xxxxx',
        lwa_client_secret='xxxxx',
        aws_secret_key='xxxxx',
        aws_access_key='xxxxx',
        role_arn='xxxxx'
    )


createReportResponse = api.Reports(credentials=credentials, marketplace=base.Marketplaces.UK).create_report(reportType='GET_XML_RETURNS_DATA_BY_RETURN_DATE', marketplaceIds=[base.Marketplaces.UK.marketplace_id])
print(createReportResponse)
```

**Reply by saleweaver** [reply]
I've just pushed some changes, can you please try if the latest works for you? This should work without passing the extra marketplace at the end

**Reply by smilekudo** [reply]
Sure let me try.

**Reply by smilekudo** [reply]
I just uninstall and install python-amazon-sp-api again but seem still have issue.
`createReportResponse = api.Reports(credentials=credentials, marketplace=base.Marketplaces.UK).create_report(reportType='GET_XML_RETURNS_DATA_BY_RETURN_DATE')`

**Reply by saleweaver** [reply]
Not pip. You would need to fork the repo or install the repo via pip...

**Reply by smilekudo** [reply]
i am new in python.
Try this and got permission denined, could you help?
`python setup.py install`

`[Errno 13] Permission denied: 'C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.7_3.7.2544.0_x64__qbz5n2kfra8p0\\Lib\\site-packages\\test-easy-install-3480.write-test'`

**Reply by smilekudo** [reply]
i have downloaded zip and extract to folder already.

**Reply by smilekudo** [reply]
Update:
- Ran successfully without passing marketplaceid.
```
# report request     
createReportResponse = api.Reports(credentials=credentials, marketplace=base.Marketplaces.UK).create_report(reportType='GET_XML_RETURNS_DATA_BY_RETURN_DATE')

```

**Reply by saleweaver** [reply]
Sorry, had a call. Good you got it working and thanks for verifying. I'll update some of the code tonight or tomorrow and will ping you once it's published.

**Reply by smilekudo** [reply]
Yes, thank you.
Will let you know if have any issues then we can make it better.

**Reply by dehidehidehi** [reply]
Thank you for troubleshooting this thread, the temporary solution worked for me.
